compaction/compaction_manager.cc

```diff
@@ -624,7 +632,9 @@ inline bool compaction_manager::can_proceed(const lw_shared_ptr<task>& task) {
 inline future<> compaction_manager::put_task_to_sleep(lw_shared_ptr<task>& task) {
     cmlog.info("compaction task handler sleeping for {} seconds",
         std::chrono::duration_cast<std::chrono::seconds>(task->compaction_retry.sleep_time()).count());
-    return task->compaction_retry.retry();
+    return task->compaction_retry.retry(task->compaction_data.abort).handle_exception_type([task] (sleep_aborted&) {
+        task->throw_compaction_stopped_exception();
```

Comment:
perhaps you missed removing task->throw_compaction_stopped_exception();? we can probably replace it with then_wrapped() and call make_compaction_stopped_exception() on sleep_aborted exception.

Reply:
sorry, I forgot to amend the patch before pushing.  Fixed now.